### DISCUSSION TITLE
Display ECI for Neighbouring Cells

### DISCUSSION THREAD

**Discussion by moffattsam** [christianrowlands/android-network-survey#62] [Q&A] [answered]
When exporting as CSV, I can see the ECI for the serving cell and all associated data. 

What I would be interested in is, is it possible to display the cell IDs for those neighbouring cells which aren't currently the serving cell, but could be at another time?

I guess in the same way that an SDR scanner would pick everything up. Or is this a limitation of SIM based on the phone that it can only collect the cell id, MNC, tac etc of the mast to which it is currently connected.

**Comment by christianrowlands** [comment, marked as answer]
Unfortunately I can't get the cell ID of a neighbor cell, at least not directly. The phone does not decode that information from the SIB message for neighbor cells to save battery and resources on the phone. Instead, it simply monitors the EARFCN, PCI, and RSRP and RSRQ values. and then if one of those becomes better than the serving cell (with some caveats), then it goes and decodes the full SIB with all the cell details to figure out if it should switch over to it. This causes the phone to use less energy as it only decodes the full cell details when it is getting ready to switch to the new tower, instead of constantly.

Now, I could keep a list of previous towers and the corresponding ECI to EARFCN/PCI mapping, and then if I see that EARFCN/PCI again try and display the ECI for the neighbor, but this would be guessing as the ECI could change, and EARFCN/PCI combos are reused throughout the network, so I would need to tie it to a geographic location as well. I don't like this idea because it would be some guess work when displaying the ECI and it could end up displaying the wrong one.

**Reply by moffattsam** [reply]
Understood and agreed, the guess work is not particularly agreeable. Must better to have less data but certainly in this regard.

Thanks for getting back to me so quickly. Love the app and it is proving very useful. 

For a free and easily accessible app it is perfect and removes the immediate necessity for an SDR option. 

Top work!
